### ISSUE TITLE
ship roles "taxi" and "trading" are not persistent

### ISSUE THREAD

**Issue by lwt-pressy** [EDCD/EDDI#818]
## EDDI version in which issue found

v3.0,2 (standalone)

## Steps to reproduce

switch ship role to "taxi" or "trading"

## Expected

roles stay per restart

## Observed

after 1rst restart the roles switch to "multipurpose" and after 2nd restart they are "combat". there it stays for the next reboots.

## Investigation

It can be reproduced. I tried it multiple times. If I switch it to Exploration it stays there.

**Comment by Tkael** [comment]
Are you switching the role of your active ship or of a random ship in your shipyard?

**Comment by lwt-pressy** [comment]
I always used the same 2 ships. For a test I just switched all my Combat ships to taxi. After restart all are Multipurpose.

**Comment by Tkael** [comment]
I've confirmed that this is occurring and that the change extends to the edname saved in the ship monitor - it isn't just a cosmetic issue with localized names.
We'll need to dig in further to see exactly why the data is being saved to the ship monitor incorrectly. Undefined roles should default to multipurpose. I'm a little baffled as to why they are switching to `Combat`.

**Comment by Tkael** [comment]
Testing this evening. I set my active ship to `Taxi` and confirmed that the data was recorded correctly in ShipMonitor.json
```
      "ShipRole": {
        "edname": "Taxi"
      },
      "role": "Taxi",
```
However, upon parsing ShipMonitor.json EDDI concluded that the ship's role was "MultiPurpose" with an edname of "Combat".
@richardbuckle You've got a better handle on this area of the code than I do. Thoughts?

**Comment by Tkael** [comment]
A related item: We don't seem to have any test coverage for data read from ShipMonitor.json?

**Comment by lwt-pressy** [comment]
Another suprising thing is that witch 3.0.3-rc1 my SLF will show up as ships in the shipyard. Is this a bug or a feature? I never seen it in previous versions.

    {
      "LocalId": 33,
      "cargocapacity": 0,
      "value": 13400,
      "rebuy": 670,
      "name": null,
      "model": "Empire_Fighter",
      "ident": "PR-28E",
      "phoneticname": null,
      "ShipRole": {
        "edname": "Combat"
      },
      "role": "Combat",
      "raw": "{ \"timestamp\":\"2018-08-31T14:48:07Z\", \"event\":\"Loadout\", \"Ship\":\"Empire_Fighter\", \"ShipID\":33, \"ShipName\":\"\", \"ShipIdent\":\"PR-28E\", \"HullValue\":5941, \"ModulesValue\":7459, \"Rebuy\":670, \"Modules\":[ { \"Slot\":\"SmallHardpoint1\", \"Item\":\"Hpt_PlasmaRepeater_Fixed_Empire_Fighter\", \"On\":true, \"Priority\":2, \"AmmoInClip\":30, \"AmmoInHopper\":30, \"Health\":1.000000 }, { \"Slot\":\"SmallHardpoint2\", \"Item\":\"Hpt_PlasmaRepeater_Fixed_Empire_Fighter\", \"On\":true, \"Priority\":2, \"AmmoInClip\":30, \"AmmoInHopper\":30, \"Health\":1.000000 }, { \"Slot\":\"TinyHardpoint1\", \"Item\":\"Hpt_ChaffLauncher_Tiny\", \"On\":true, \"Priority\":2, \"AmmoInClip\":1, \"AmmoInHopper\":10, \"Health\":1.000000, \"Value\":7459 }, { \"Slot\":\"Armour\", \"Item\":\"Empire_Fighter_Armour_Standard\", \"On\":true, \"Priority\":2, \"Health\":1.000000 }, { \"Slot\":\"VesselVoice\", \"Item\":\"VoicePack_Verity\", \"On\":true, \"Priority\":2, \"Health\":1.000000 }, { \"Slot\":\"ShipCockpit\", \"Item\":\"Empire_Fighter_Cockpit\", \"On\":true, \"Priority\":2, \"Health\":1.000000 } ] }",
      "RawIsNotNull": true,
      "starsystem": null,
      "station": null,
      "health": 100.0,
      "cargohatch": null,

**Comment by Tkael** [comment]
Bug I think. It looks like you managed to trigger a loadout event while you were in a fighter and it wrote your fighter into the shipyard as if it was a complete ship. Did you log out and log back in from your fighter at some point perhaps?

**Comment by lwt-pressy** [comment]
I don't think so. I only used the figher myself for a few test drives. While in combat I only used it for my npc pilot. The only unusual event I could think of is that I once jumb to supercruise and forgot to retrieve the figther so it got destroyed. I only paste the one figther above from shipyard. In the shipyard both of my fighter types are shown. I also have a new trident.

Would it help if I search my ED logfiles for something?

**Comment by lwt-pressy** [comment]
might this be something?

after a dock (im not sure i launched it again:

`{ "timestamp":"2018-08-31T14:47:54Z", "event":"DockFighter" }

{ "timestamp":"2018-08-31T14:48:04Z", "event":"FighterRebuilt", "Loadout":"one" }

{ "timestamp":"2018-08-31T14:48:05Z", "event":"LaunchFighter", "Loadout":"two", "PlayerControlled":true }

{ "timestamp":"2018-08-31T14:48:07Z", "event":"Loadout", "Ship":"Empire_Fighter", "ShipID":33, "ShipName":"", "ShipIdent":"PR-28E", "HullValue":5941, "ModulesValue":7459, "Rebuy":670, "Modules":[ { "Slot":"SmallHardpoint1", "Item":"Hpt_PlasmaRepeater_Fixed_Empire_Fighter", "On":true, "Priority":2, "AmmoInClip":30, "AmmoInHopper":30, "Health":1.000000 }, { "Slot":"SmallHardpoint2", "Item":"Hpt_PlasmaRepeater_Fixed_Empire_Fighter", "On":true, "Priority":2, "AmmoInClip":30, "AmmoInHopper":30, "Health":1.000000 }, { "Slot":"TinyHardpoint1", "Item":"Hpt_ChaffLauncher_Tiny", "On":true, "Priority":2, "AmmoInClip":1, "AmmoInHopper":10, "Health":1.000000, "Value":7459 }, { "Slot":"Armour", "Item":"Empire_Fighter_Armour_Standard", "On":true, "Priority":2, "Health":1.000000 }, { "Slot":"VesselVoice", "Item":"VoicePack_Verity", "On":true, "Priority":2, "Health":1.000000 }, { "Slot":"ShipCockpit", "Item":"Empire_Fighter_Cockpit", "On":true, "Priority":2, "Health":1.000000 } ] }
`

Or when I launched a Fighter (just for testing, driving it myself):

`{ "timestamp":"2018-08-31T14:38:29Z", "event":"LaunchFighter", "Loadout":"one", "PlayerControlled":true }

{ "timestamp":"2018-08-31T14:38:31Z", "event":"Loadout", "Ship":"GDN_Hybrid_Fighter_V1", "ShipID":34, "ShipName":"", "ShipIdent":"PR-31G", "Rebuy":670, "Modules":[ { "Slot":"GDN_Hybrid_Fighter_V1_Shield", "Item":"Ext_Emitter_Guardian", "On":true, "Priority":2, "Health":1.000000 }, { "Slot":"SmallHardpoint1", "Item":"Hpt_GuardianPlasma_Fixed_GDN_Fighter", "On":true, "Priority":2, "Health":1.000000 }, { "Slot":"ShieldGenerator", "Item":"Int_ShieldGenerator_Fighter_Class1", "On":true, "Priority":2, "Health":1.000000 }, { "Slot":"Armour", "Item":"GDN_Hybrid_Fighter_V1_Armour_Standard", "On":true, "Priority":2, "Health":1.000000 }, { "Slot":"VesselVoice", "Item":"VoicePack_Verity", "On":true, "Priority":2, "Health":1.000000 }, { "Slot":"ShipCockpit", "Item":"GDN_Hybrid_Fighter_V1_Cockpit", "On":true, "Priority":2, "Health":1.000000 } ] }
`

**Comment by lwt-pressy** [comment]
It looks like that always when you use the fighter yourself ( "PlayerControlled":true ) a loadout event follows a LaunchFighter.

**Comment by Tkael** [comment]
That's interesting. I'm pretty sure that's a new behavior.

**Comment by Hoodathunk** [comment]
PR #882 submitted

**Comment by richardbuckle** [comment]
Fixed for 3.1-b3 by #887.
